src/libasr/ASR.asdl

```diff
@@ -266,6 +270,7 @@ expr
     | ArrayPack(expr array, expr mask, expr? vector, ttype type, expr? value)
     | Transfer(expr source, expr mold, expr? size, ttype type, expr? value)
     | DerivedRef(expr v, symbol m, ttype type, expr? value)
+    | OverloadedCompare(expr left, cmpop op, expr right, ttype type, expr? value, expr overloaded)
```

Comment:
I am -1 on putting overloaded compare as a separate node. An extra attribute in `Compare` was enough I think.

Reply:
We don't have Compare anymore.

So which of the IntegerCompare, RealCompare etc should have the extra member? Note that neither Python nor Fortran allows you to overload Integer or Real comparison. It allows you to overload a user defined type compare, which is what the OverloadedCompare node does.

Reply:
No issues. Let’s have it then.